app/app_iot_core.go

```diff
@@ -26,22 +26,28 @@ import (
 	"github.com/mendersoftware/iot-manager/model"
 )
 
+func getIoTCoreSession(integration model.Integration) (*session.Session, error) {
+	accessKeyID := integration.Credentials.AccessKeyID
+	secretAccessKey := integration.Credentials.SecretAccessKey
+	if accessKeyID == nil || secretAccessKey == nil {
+		return nil, ErrNoCredentials
+	}
+	return session.NewSession(&aws.Config{
+		Credentials: credentials.NewStaticCredentials(*accessKeyID, *secretAccessKey, ""),
+	})
```

Comment:
I disagree, this is part of the contract we sign with the AWS SDK: it either return a session, or an error.
Returning ErrCannotAllocateSession will cause 500, which is exactly the same when we panic.
What's the advantage of returning a custom error? It will only mask the actual issue.